src/go/rpk/pkg/config/config_test.go

```diff
@@ -793,15 +793,22 @@ rpk:
 			name: "shall write config with archival configuration",
 			conf: func() *Config {
 				c := getValidConfig()
-				c.Redpanda.ArchivalStorage = ArchivalStorage{
-					Enabled:                  true,
-					S3AccessKey:              "access",
-					S3SecretKey:              "secret",
-					S3Region:                 "region",
-					S3Bucket:                 "bucket",
-					ReconciliationIntervalMs: 3,
-					MaxConnections:           4,
-				}
+				enabled := true
+				access := "access"
+				bucket := "bucket"
+				region := "region"
+				secret := "secret"
+				interval := 20000
+				conns := 4
+				endpoint := "http"
+				c.Redpanda.ArchivalStorageEnabled = &enabled
+				c.Redpanda.ArchivalStorageS3AccessKey = &access
```

Comment:
I would like to see one more test that when you don't provide e.g. max connections, that won't be included in the conf so it will be defaulted on the redpanda side but ... maybe it's obvious that would work :)

Reply:
Nope, I think it's also worth testing that users who don't need this won't be seeing it written to the config file (basically a afeguard against removing the `omitempty`).

Reply:
The case where no archival fields are provided is covered by (any of) the remaining tests that do not expected archival in the output. Right?

Are you thinking of a mixed test scenario, where some archival fields are provide and some not?

Reply:
Ah, you're totally right. This is enough then.